lms/views/api/blackboard/files.py

```diff
@@ -4,72 +4,38 @@
 from pyramid.view import view_config, view_defaults
 
 from lms.security import Permissions
-from lms.services import HTTPError
 from lms.views import helpers
-from lms.views.api.blackboard._schemas import (
-    BlackboardListFilesSchema,
-    BlackboardPublicURLSchema,
-)
-from lms.views.api.blackboard.exceptions import BlackboardFileNotFoundInCourse
 
 #: A regex for parsing just the file_id part out of one of our custom
 #: blackboard://content-resource/<file_id>/ URLs.
 DOCUMENT_URL_REGEX = re.compile(
     r"blackboard:\/\/content-resource\/(?P<file_id>[^\/]*)\/"
 )
 
-# The maxiumum number of paginated requests we'll make before returning.
-PAGINATION_MAX_REQUESTS = 25
-
-# The maximum number of results to request per paginated response.
-# 200 is the highest number that Blackboard will accept here.
-PAGINATION_LIMIT = 200
-
 
 @view_defaults(permission=Permissions.API, renderer="json")
 class BlackboardFilesAPIViews:
     def __init__(self, request):
         self.request = request
-        self.blackboard_api_client = request.find_service(
-            name="basic_blackboard_api_client"
-        )
+        self.blackboard_api_client = request.find_service(name="blackboard_api_client")
 
     @view_config(request_method="GET", route_name="blackboard_api.courses.files.list")
     def list_files(self):
         """Return the list of files in the given course."""
-        course_id = self.request.matchdict["course_id"]
 
-        files = []
-        path = f"courses/uuid:{course_id}/resources?type=file&limit={PAGINATION_LIMIT}"
-
-        for _ in range(PAGINATION_MAX_REQUESTS):
-            response = self.blackboard_api_client.request("GET", path)
-            files.extend(BlackboardListFilesSchema(response).parse())
-            path = response.json().get("paging", {}).get("nextPage")
-            if not path:
-                break
-
-        return files
+        return self.blackboard_api_client.list_files(
+            self.request.matchdict["course_id"]
+        )
```

Comment:
There's no transformation for this view to do to the `list_files()` response because the schema is still doing it all. That'll move from the schema into this view later